xdsl/dialects/builtin.py

```diff
@@ -636,6 +636,9 @@ def __eq__(self, other: Any):
             math.isnan(self.data) and math.isnan(other.data) or self.data == other.data
         )
 
+    def __hash__(self):
+        return super().__hash__()
```

Comment:
Does this work? Would cut out the middle man
```suggestion
        return hash(self.data)
```

Reply:
It does indeed